Jenkinsfile

```diff
@@ -0,0 +1,399 @@
+        Finally archive the results.
+        */
+        /*
+        stage('Update Beats') { 
+            agent { label 'linux' }
+            
+            steps {
+              ansiColor('xterm') {
+                  deleteDir()
+                  dir("${BASE_DIR}"){
+                    unstash 'source'
+                    sh """
+                    #!
+                    ./script/jenkins/update-beats.sh
+                    """
+                    archiveArtifacts allowEmptyArchive: true, artifacts: "${BASE_DIR}/build", onlyIfSuccessful: false
+                  }
+                }
+              }
+        }*/
+      }
+    }
+    /**
+    Build the documentation and archive it.
+    Finally archive the results.
+    */
+    stage('Documentation') { 
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+        HOME = "${env.WORKSPACE}"
+        GOPATH = "${env.WORKSPACE}"
+      }
+      when { 
+        beforeAgent true
+        allOf {
+          anyOf {
+            branch 'master'
+            environment name: 'Run_As_Master_Branch', value: 'true'
+          }
+          environment name: 'doc_ci', value: 'true' 
+        }
+      }
+      steps {
+        withEnvWrapper() {
+          unstash 'source'
+          dir("${BASE_DIR}"){  
+            sh """#!/bin/bash
+            set -euxo pipefail
+            make docs
+            """
+          }
+        }
+      }
+      post{
+        success {
+          tar(file: "doc-files.tgz", archive: true, dir: "html_docs", pathPrefix: "${BASE_DIR}/build")
+        }
+      }
+    }
+    /**
+      build release packages.
+    */
+    stage('Release') { 
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      when { 
+        beforeAgent true
+        allOf {
+          anyOf {
+            not {
+              changeRequest()
+            }
+            branch 'master'
+            branch "\\d+\\.\\d+"
+            branch "v\\d?"
+            tag "v\\d+\\.\\d+\\.\\d+*"
+            environment name: 'Run_As_Master_Branch', value: 'true'
+          }
+          environment name: 'releaser_ci', value: 'true' 
+        }
+      }
+      steps {
+        withEnvWrapper() {
+          unstash 'source'
+          dir("${BASE_DIR}"){
+            sh './script/jenkins/package.sh'
+          }
+        }  
+      }
+      post {
+        success {
+          echo "Archive packages"
+          /** TODO check if it is better storing in snapshots */
+          //googleStorageUpload bucket: "gs://${JOB_GCS_BUCKET}/${JOB_NAME}/${BUILD_NUMBER}", credentialsId: "${JOB_GCS_CREDENTIALS}", pathPrefix: "${BASE_DIR}/build/distributions/", pattern: '${BASE_DIR}/build/distributions//**/*', sharedPublicly: true, showInline: true
+        }
+      }
+    }
+    /**
+    Checks if kibana objects are updated.
+    */
+    stage('Check kibana Obj. Updated') { 
```

Comment:
Please run this before the `Release` step.

Reply:
ok